artemis-journal/src/main/java/org/apache/activemq/artemis/core/io/AbstractSequentialFile.java

```diff
@@ -91,7 +92,10 @@ public final void delete() throws IOException, InterruptedException, ActiveMQExc
          close();
       }
 
-      if (file.exists() && !file.delete()) {
+      try {
+         Files.deleteIfExists(file.toPath());
```

Comment:
It is helpful in difference cases: 

- NFS/GlusterFS volumes because it avoids 2 rount-trip across network
- local storage: is atomic with other file system operations and much cheaper re native (eg malloc) allocations on JNI side